File: freelance-system/src/main/java/com/example/freelance_system/model/FreelancerProfile.java
package com.example.freelance_system.model;
import jakarta.persistence.*;

@Entity
@Table(name = "freelancer_profiles")
public class FreelancerProfile {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private String title;
    private String overview;
    private String skills;
    private String education;
    private String experience;
    private String hourlyRate;
    private String profileImageUrl;
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getUserId() {
		return userId;
	}
	public void setUserId(Long userId) {
		this.userId = userId;
	}
	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	public String getOverview() {
		return overview;
	}
	public void setOverview(String overview) {
		this.overview = overview;
	}
	public String getSkills() {
		return skills;
	}
	public void setSkills(String skills) {
		this.skills = skills;
	}
	public String getEducation() {
		return education;
	}
	public void setEducation(String education) {
		this.education = education;
	}
	public String getExperience() {
		return experience;
	}
	public void setExperience(String experience) {
		this.experience = experience;
	}
	public String getHourlyRate() {
		return hourlyRate;
	}
	public void setHourlyRate(String hourlyRate) {
		this.hourlyRate = hourlyRate;
	}
	public String getProfileImageUrl() {
		return profileImageUrl;
	}
	public void setProfileImageUrl(String profileImageUrl) {
		this.profileImageUrl = profileImageUrl;
	}


}

File: freelance_app/lib/models/freelancer_profile.dart
class FreelancerProfile {
  int? id;
  int userId;
  String? title;
  String? overview;
  String? skills;
  String? education;
  String? experience;
  String? hourlyRate;
  String? profileImageUrl;

  FreelancerProfile({
    this.id,
    required this.userId,
    this.title,
    this.overview,
    this.skills,
    this.education,
    this.experience,
    this.hourlyRate,
    this.profileImageUrl,
  });

  factory FreelancerProfile.fromJson(Map<String, dynamic> json) => FreelancerProfile(
        id: json['id'],
        userId: json['userId'],
        title: json['title'],
        overview: json['overview'],
        skills: json['skills'],
        education: json['education'],
        experience: json['experience'],
        hourlyRate: json['hourlyRate'],
        profileImageUrl: json['profileImageUrl'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'userId': userId,
        'title': title,
        'overview': overview,
        'skills': skills,
        'education': education,
        'experience': experience,
        'hourlyRate': hourlyRate,
        'profileImageUrl': profileImageUrl,
      };
}
